src/Ocelot.Provider.Consul/ConsulProviderFactory.cs

```diff
@@ -17,16 +18,20 @@ public static class ConsulProviderFactory
 
     public static ServiceDiscoveryFinderDelegate Get { get; } = CreateProvider;
 
+    private static ConsulRegistryConfiguration configuration;
+    private static ConsulRegistryConfiguration ConfigurationGetter() => configuration;
+    public static Func<ConsulRegistryConfiguration> GetConfiguration { get; } = ConfigurationGetter;
+
     private static IServiceDiscoveryProvider CreateProvider(IServiceProvider provider,
         ServiceProviderConfiguration config, DownstreamRoute route)
     {
         var factory = provider.GetService<IOcelotLoggerFactory>();
         var consulFactory = provider.GetService<IConsulClientFactory>();
 
-        var consulRegistryConfiguration = new ConsulRegistryConfiguration(
-            config.Scheme, config.Host, config.Port, route.ServiceName, config.Token);
+        configuration = new ConsulRegistryConfiguration(config.Scheme, config.Host, config.Port, route.ServiceName, config.Token);
```

Comment:
Why not foresee a default dto for the configuration, instead of passing the arguments like that?

Reply:
Apologies, that design isn't mine. 😮 It's Tom's creation! 😄
I considered refactoring the `ConsulProviderFactory` and `Consul` classes, but it was outside the bug's/PR scope. Therefore, I've only made changes to ensure the successful injection of the configuration into the `DefaultConsulServiceBuilder` constructor. We can add a TODO task for further refactoring in future iterations.

Reply:
@raman-m Ok, fine just add a TODO for that matter :-)

Reply:
Creating new TODOs poses no issues, but the real fun begins after this PR is merged... 🤣